torch/lib/THD/base/data_channels/DataChannelMPI.cpp

```diff
@@ -93,7 +93,13 @@ DataChannelMPI::~DataChannelMPI() {
 
 
 bool DataChannelMPI::init() {
-  MPI_Init(NULL, NULL);
+  int* provided = NULL;
+  MPI_Init_thread(NULL, NULL, MPI_THREAD_MULTIPLE, provided);
+  if (*provided != MPI_THREAD_MULTIPLE) {
+    std::cerr << "MPI implementation does not support multiple threads."
```

Comment:
WARNING: Used MPI implementation doesn't support multithreading, so distributed functions might not work properly.